modelling/src/main/java/org/axonframework/modelling/command/inspection/AnnotatedAggregate.java

```diff
@@ -409,11 +404,23 @@ private Object handle(CommandMessage<?> commandMessage) throws Exception {
                 inspector.commandHandlerInterceptors((Class<? extends T>) aggregateRoot.getClass())
                          .map(chi -> new AnnotatedCommandHandlerInterceptor<>(chi, aggregateRoot))
                          .collect(Collectors.toList());
+        AtomicBoolean hasHandler = new AtomicBoolean(false);
         MessageHandlingMember<? super T> handler =
                 inspector.commandHandlers((Class<? extends T>) aggregateRoot.getClass())
-                         .filter(mh -> mh.canHandle(commandMessage))
+                         .filter(mh -> hasHandler.compareAndSet(mh.canHandle(commandMessage),true)
```

Comment:
I tried to fix this by filter handlers that can handle the command and collect it into the list (if empty throw **NoHandlerForCommandException**) then find the handlers that can resolve the target again (if can not resolve then throw **AggregateEntityNotFoundException**). Am I doing it right, Thanks for the feedback.

Reply:
Hi, at first glance, this approach seems correct. I say "first glance" but needed to look at it quite thoroughly 😉  . 
I am wondering if it would make the code a bit more readable to assign the "potential" handlers to a list first. If that list is empty, we can't handle the command at all. If the list is not empty, we will find a suitable handler to forward it to considering the state of the aggregate. If the resulting list is then empty, the other exception is thrown.

Reply:
Ok I fixed it to be more readable, please take a look.